store/copr/mpp.go

```diff
@@ -244,6 +265,7 @@ func (m *mppIterator) establishMPPConns(bo *tikv.Backoffer, req *kv.MPPDispatchR
 	rpcResp, err := m.store.GetTiKVClient().SendRequest(bo.GetCtx(), req.Meta.GetAddress(), wrappedReq, tikv.ReadTimeoutUltraLong)
 
 	if err != nil {
+		m.cancelMppTask(bo, taskMeta)
```

Comment:
If a task can return error successfully, it's unnecessary to cancal it again. Instead, cancelling the sent tasks makes more sense.

Reply:
if a task returns errors, we should continue to cancel the tasks on the same server and on other servers.  Even if sending unnecessary cancel cmd, it does not result into exceptions. 


Instead, cancelling the sent tasks makes more sense. ->
I add detecting the kill when dispatching tasks.

Reply:
I got it but still have concerns now. Firstly, the name of the function makes people confused that it works for one task or all the tasks. Secondly, If there are multiple failures, we would send the cancels for multiple times, this is a waste although it will not always cause real problems. Thirdly, we cannot ganrantee that the stores in the cache are stable. That might be less than the time we made the plans. Now that we have know all the stores involved in this query, why must we fetch the stores in the cache?

Reply:
the stores are updated.